profiler/test/Datadog.Profiler.IntegrationTests/SingleStepInstrumentation/SingleStepInstrumentationTest.cs

```diff
@@ -0,0 +1,463 @@
+
+            agent.NbCallsOnProfilingEndpoint.Should().NotBe(0);
+        }
+
+        [TestAppFact("Samples.BuggyBits")]
+        public void CheckSsiDeployedAndProfilingSsiEnabled_Delayed(string appName, string framework, string appAssembly)
+        {
+            var runner = new TestApplicationRunner(appName, framework, appAssembly, _output, commandLine: "--scenario 1", enableProfiler: false, enableTracer: true);
+
+            // deployed with SSI
+            runner.Environment.SetVariable(EnvironmentVariables.SsiDeployed, "profiler");
+            // simulate long lived
+            runner.Environment.SetVariable(EnvironmentVariables.SsiShortLivedThreshold, TimeSpan.FromSeconds(6).TotalMilliseconds.ToString());
+
+            using var agent = MockDatadogAgent.CreateHttpAgent(_output);
+
+            List<Serie> nbProfilesSeries = [];
+            Serie runtimeIdSerie = null;
+            agent.TelemetryMetricsRequestReceived += (_, ctx) =>
+            {
+                var s = GetRequestText(ctx.Value.Request);
+
+                var (nbRuntimeId, nbProfiles) = ExtractSeries(s);
+
+                if (nbRuntimeId != null && runtimeIdSerie != null)
+                {
+                    Assert.Fail("There must be only one 'ssi_heuristic.number_of_runtime_id' serie'");
+                }
+
+                runtimeIdSerie = nbRuntimeId;
+                nbProfilesSeries.AddRange(nbProfiles);
+            };
+
+            runner.Run(agent);
+
+            runtimeIdSerie.Should().NotBeNull();
+            runtimeIdSerie.Tags.Should().Contain("has_sent_profiles:true", "heuristic_hypothetical_decision:triggered", "installation:ssi", "enablement_choice:ssi_enabled");
+
+            nbProfilesSeries.Should().NotBeEmpty();
+            var hasNotSentProfiles = false;
+            var hasTriggeredSeries = false;
+            nbProfilesSeries.Should().AllSatisfy(s =>
+            {
+                s.Metric.Should().Be("ssi_heuristic.number_of_profiles");
+                s.Tags.Should().HaveCount(4).And.OnlyHaveUniqueItems().And.Contain("installation:ssi", "enablement_choice:ssi_enabled");
+                // we need to check for no_span in case of the timer finished after the first call (flakiness)
+                if (s.Tags.Contains("heuristic_hypothetical_decision:short_lived") || s.Tags.Contains("heuristic_hypothetical_decision:no_span"))
+                {
+                    s.Tags.Should().Contain("has_sent_profiles:false");
+                    hasNotSentProfiles = true;
+                }
+                else if (s.Tags.Contains("heuristic_hypothetical_decision:triggered"))
+                {
+                    s.Tags.Should().Contain("has_sent_profiles:true");
+                    hasTriggeredSeries = true;
+                }
+                else
+                {
+                    Assert.Fail("Unrecognized heuristic_hypothetical_decision");
```

Comment:
is there a way to provide the unexpected value in the assert so we could see it if it happens?